File: EDAII x Java/Ordenamiento/MergeSort.java
public class MergeSort{
	int[] arr = {33,50,17,48,23,8,60};

	public static void main(String[] args){
		MergeSort ms = new MergeSort();
		ms.mergeSort(ms.arr, 0, ms.arr.length-1);

		for (int i=0; i<ms.arr.length; i++){
			System.out.print(ms.arr[i] + " ");
		}
	}

	public void mergeSort(int[] A, int inf, int sup){
		if(inf==sup-1){ // Dos elementos
			if(A[inf]>A[sup]){
				int tmp = A[inf];
				A[inf]=A[sup];
				A[sup]=tmp;
			}
		} else if(inf < sup){
			int med = (inf+sup)/2;

			mergeSort(A,inf,med);
			mergeSort(A,med+1,sup);
			merge(A,inf,med,sup);
		}
	}

	public void merge(int[] A, int inf, int med, int sup){
		int n1 = med-inf+1;
		int n2 = sup-med;

		int L[] = new int[n1];
		int R[] = new int[n2];

		// Copiando a los arreglos auxiliares
		for(int i=0; i<n1; i++){
			L[i]=A[inf+i];
		}
		for(int j=0; j<n2; j++){
			R[j]=A[med+1+j];
		}

		int i=0, j=0;
		int k=inf;

		// Obteniendo valores menores de ambos arreglos auxiliares
		while(i<n1 && j<n2){
			if(L[i]<=R[j]){
				A[k]=L[i];
				i++;
			} else{
				A[k]=R[j];
				j++;
			}
			k++;
		}

		// Copiando el resto de valores no asignados
		while(i<n1){
			A[k]=L[i];
			i++;
			k++;
		}

		while(j<n2){
			A[k]=R[j];
			j++;
			k++;
		}
	}

}

File: EDAII x Python/Ordenamiento/MergeSort.py
class MergeSort:

	arr = []

	def __init__(self, arr):
		'Constructor de la clase.'
		self.arr = arr

	def mergesort(self, inf, sup):
		'Algoritmo de ordenamiento.'
		if inf==(sup-1): # Dos elementos
			if self.arr[inf] > self.arr[sup]:
				tmp = self.arr[inf];
				self.arr[inf] = self.arr[sup];
				self.arr[sup] = tmp;
		elif inf < sup:
			med = int((inf+sup)/2)

			self.mergesort(inf, med)
			self.mergesort(med+1, sup)
			self.merge(inf, med, sup)

	def merge(self, inf, med, sup):
		'Mezcla las dos sublistas de manera ordenada.'
		n1 = med-inf+1
		n2 = sup-med

		L = [0] * n1
		R = [0] * n2

		# Copiando a los arreglos auxiliares
		for i in range(0,n1):
			L[i] = self.arr[inf+i]

		for j in range(0,n2):
			R[j] = self.arr[med+1+j]

		i = j = 0
		k = inf

		# Obteniendo valores menores de ambos arreglos auxiliares
		while i<n1 and j<n2:
			if L[i]<= R[j]:
				self.arr[k] = L[i]
				i+=1
			else:
				self.arr[k] = R[j]
				j+=1
			k+=1

		# Copiando el resto de valores no asignados
		while i<n1:
			self.arr[k] = L[i]
			i+=1
			k+=1

		while j<n2:
			self.arr[k] = R[j]
			j+=1
			k+=1

# Ejecutando el objeto
programa = MergeSort([33,50,17,48,23,8,60])
programa.mergesort(0, len(programa.arr)-1)
print(programa.arr)